### ISSUE TITLE
coffee-script/register test fails on win8.0 x64 node 0.10.32

### ISSUE THREAD

**Issue by iki** [tkellen/node-liftoff#23]
```
> mocha -R spec test/index.js
  1) Liftoff buildEnvironment should attempt pre-loading local modules based on extension option:
     AssertionError: expected undefined to equal 'coffee-script/register'
    at Context.<anonymous> (C:\Users\iki\..js\node-liftoff\test\index.js:54:23)
    at callFn (C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runnable.js:249:21)
    at Test.Runnable.run (C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runnable.js:242:7)
    at Runner.runTest (C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runner.js:373:10)
    at C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runner.js:451:12
    at next (C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runner.js:298:14)
    at C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runner.js:308:7
    at next (C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runner.js:246:23)
    at Object._onImmediate (C:\Users\iki\AppData\Roaming\npm\node_modules\mocha\lib\runner.js:275:5)
    at processImmediate [as _immediateCallback] (timers.js:345:15)
```
Discovered when using gulp with gulpfile.coffee, and originally reported at https://github.com/gulpjs/gulp/issues/717.

**Comment by dogancelik** [comment]
I have this issue too. I can't use gulp 3.8 with gulpfile.coffee.

Node version: 0.10.33
Windows version: 7 (64-bit)

```
$ npm test
36 passing (753ms)
1 failing

1) Liftoff buildEnvironment should attempt pre-loading local modules based on extension option:
   AssertionError: expected undefined to equal 'coffee-script/register'
  at Context.<anonymous> (D:\07. Alınan Dosyalarım\node-liftoff-master\test\index.js:57:23)
  at Test.Runnable.run (D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runnable.js:221:32)
  at Runner.runTest (D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runner.js:374:10)
  at D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runner.js:452:12
  at next (D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runner.js:299:14)
  at D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runner.js:309:7
  at next (D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runner.js:247:23)
  at Object._onImmediate (D:\07. Alınan Dosyalarım\node-liftoff-master\node_modules\mocha\lib\runner.js:276:5)
  at processImmediate [as _immediateCallback] (timers.js:345:15)
```

**Comment by tkellen** [comment]
which shell are you using?

**Comment by dogancelik** [comment]
@tkellen I'm using Command Prompt

**Comment by tkellen** [comment]
can you confirm if the issue exists using powershell and or git shell?

**Comment by dogancelik** [comment]
Yup issue exists on both.

![test](https://cloud.githubusercontent.com/assets/486818/5607408/a32d818e-9463-11e4-9d45-8b93a0090380.jpg)

**Comment by tkellen** [comment]
I'm not able to reproduce this on my test machine.  Can you add `console.log(configPath)` to line 111 of index.js and share the output?

**Comment by dogancelik** [comment]
@tkellen Yes! I found the problem, it's the regex that breaks the things.

Tl;DR FIX:

```js
var configExtension = /(\.[^\/]*)?$/.exec(configPath)[0];
```
You see this code will only get everything except `/` but this will work for a Unix path because `/` is used for Unix systems. But Windows uses `\` so it won't work for a Windows path.

New regex:
```js
var configExtension = /(\.[^\/\\]*)?$/.exec(configPath)[0];
```

**If I add `\\` to the regex, test works perfectly (in `D:\07. Alınan Dosyalarım\node-liftoff-master`).
But test breaks in `C:\node-liftoff-master` with symlink error**

Logging code after ``var configExtension`:
```js
console.log(configPath);
console.log(configExtension);
```

When my cwd: `D:\07. Alınan Dosyalarım\node-liftoff-master` (has spaces)
```
D:\07. Alınan Dosyalarım\node-liftoff-master\test\fixtures\coffee\mochafile.coffee
. Alınan Dosyalarım\node-liftoff-master\test\fixtures\coffee\mochafile.coffee
```

When my cwd is: `C:\node-liftoff-master`
```
C:\node-liftoff-master\test\fixtures\coffee\mochafile.coffee
.coffee
C:\node-liftoff-master\test\fixtures\coffee\mochafile.coffee.md
.coffee.md
```

**Comment by tkellen** [comment]
Awesome!  Would you mind submitting a PR?

**Comment by tkellen** [comment]
Also, could you add a test case for this so we can ensure it never crops up again?

**Comment by dogancelik** [comment]
Alright I'll try

**Comment by dogancelik** [comment]
@tkellen Wouldn't it be more safe to use the regex on just the file name (`mochafile.coffee`) rather than the whole file path (`C:\node-liftoff-master\test\fixtures\coffee\mochafile.coffee`).

Like this (indented for readability):
```js
var configExtension = /(\.[^\/\\]*)?$/.exec(
    require('path').basename(configPath)
)[0];
```

**Comment by tkellen** [comment]
It sure would!  I'd take a PR for that, and no need for the additional test case since it would become irrelevant.

**Comment by tkellen** [comment]
Resolved in df3630f58836cffbb0bbf8f967932d4484ac20c5
